File: Code/new_fitervari_backend/src/main/java/at/htl/model/Customer.java
package at.htl.model;


import javax.persistence.*;
import java.time.LocalDate;

@Entity
@Table(name = "customer")
public class Customer extends Person {
    @Column(name = "join_date")
    private LocalDate joinDate;

    @ManyToOne(cascade = CascadeType.MERGE)
    @JoinColumn(name = "trainer_id")
    private Trainer myTrainer;

    @Column(name = "cash_customer")
    private boolean CashCustomer;

    @Column(name = "member_till")
    private LocalDate memberTill;

    //region GetterSetter
    public LocalDate getJoinDate() {
        return joinDate;
    }

    public void setJoinDate(LocalDate joinDate) {
        this.joinDate = joinDate;
    }

    public Trainer getMyTrainer() {
        return myTrainer;
    }

    public void setMyTrainer(Trainer myTrainer) {
        this.myTrainer = myTrainer;
    }

    public boolean isCashCustomer() {
        return CashCustomer;
    }

    public void setCashCustomer(boolean cashCustomer) {
        CashCustomer = cashCustomer;
    }

    public LocalDate getMemberTill() {
        return memberTill;
    }

    public void setMemberTill(LocalDate memberTill) {
        this.memberTill = memberTill;
    }
    //endregion
}

File: Code/fitervari_backend/src/main/java/at/htl_leonding/model/Customer.kt
package at.htl_leonding.model

import java.util.*
import javax.persistence.*

@Entity
@Table(name = "customer")
class Customer: Person() {

    @Column(name = "join_date")
    lateinit var joinDate: Date

    @ManyToOne(cascade = arrayOf(CascadeType.MERGE))
    @JoinColumn(name = "trainer_id")
    lateinit var myTrainer: Trainer

    @Column(name = "cash_customer")
    var cashCostumer: Boolean = false

    @Column(name = "member_till")
    lateinit var memberTill: Date


    fun copyValues(other: Customer) {
        this.joinDate = other.joinDate
        this.myTrainer = other.myTrainer
        this.cashCostumer = other.cashCostumer
        this.memberTill = other.memberTill
        this.pictureId = other.pictureId
    }
}